packages/components/src/components/common/y-axis/as-y-axis.tsx

```diff
@@ -0,0 +1,80 @@
+import { Component, Element, Prop } from '@stencil/core';
+import { axisLeft } from 'd3-axis';
+import { scaleLinear } from 'd3-scale';
+import { select, Selection } from 'd3-selection';
+import readableNumber from '../../../utils/readable-number';
+
+
+/**
+ * Helper class to draw the vertical axis on some widgets.
+ * WARNING: This component should be placed next to the SVG element.
+ *
+ * @export
+ * @class YAxis
+ */
+@Component({
+  shadow: false,
+  styleUrl: './as-y-axis.scss',
+  tag: 'as-y-axis',
+})
+export class YAxis {
+
+  /**
+   * Lower limit of the axis
+   *
+   * @type {number}
+   * @memberof YAxis
+   */
+  @Prop() public from: number = 0;
+
+  /**
+   * Upper limit of the axis
+   *
+   * @type {Number[]}
+   * @memberof YAxis
+   */
+  @Prop() public to: number = 0;
+
+
+  /**
+   * Reference to the web component element.
+   */
+  @Element() private element: HTMLElement;
+
+
+  public render() {
+    const VERTICAL_SPACING = 36; // Need 16px top and bottom to view the labels
+    const TICK_RIGHT_MARGIN = 18;
+    const LABEL_WIDTH = 25;
+    const ELEMENT = select(this.element.previousElementSibling);
```

Comment:
I really like the idea of having this as an external thing. I don't like the adjacency requirement though. What if you want an x-axis component as well? You'd have to make a more generic as-axis, with extra props.

Would it be possible to pass the component ref from outside? As a prop. 

Or maybe as a hoc?

`<as-axis><as-stackedbar-widget/></as-axis>`

Moreover, does this have to be a Stencil Component? Couldn't it be a function?

Reply:
yep, I'll take a look for a better approach